aspnetcore/host-and-deploy/windows-service/scripts/2.x/RegisterService.ps1

```diff
@@ -16,12 +16,10 @@ param(
     $User
```

Comment:
I assume this is supposed to be `[string]`

Reply:
Yes, they all are `[string]`. *Forgive my horrible PS skills!* Too many languages ... not enough days in the week.

Reply:
Like this then ...

```powershell
param(
    [Parameter(mandatory=$true)]
    [String[]]
    $Name,
    [Parameter(mandatory=$true)]
    [String[]]
    $DisplayName,
    [Parameter(mandatory=$true)]
    [String[]]
    $Description,
    [Parameter(mandatory=$true)]
    $Path,
    [Parameter(mandatory=$true)]
    [String[]]
    $Exe,
    [Parameter(mandatory=$true)]
    [String[]]
    $User
)
```

Reply:
```suggestion
    [string]
    $User
```

Reply:
```powershell
param(
    [Parameter(mandatory=$true)]
    [String]
    $Name,
    [Parameter(mandatory=$true)]
    [String]
    $DisplayName,
    [Parameter(mandatory=$true)]
    [String]
    $Description,
    [Parameter(mandatory=$true)]
    $Path,
    [Parameter(mandatory=$true)]
    [String]
    $Exe,
    [Parameter(mandatory=$true)]
    [String]
    $User
)
```

Passing them as arrays would not have the expected results

Reply:
Ah ... gotcha ...

```powershell
param(
    [Parameter(mandatory=$true)]
    [string]
    $Name,
    [Parameter(mandatory=$true)]
    [string]
    $DisplayName,
    [Parameter(mandatory=$true)]
    [string]
    $Description,
    [Parameter(mandatory=$true)]
    [string]
    $Path,
    [Parameter(mandatory=$true)]
    [string]
    $Exe,
    [Parameter(mandatory=$true)]
    [string]
    $User
)
```

Note that the PS docs go with `[String[]]`.

Reply:
`$exe` could be `[System.IO.FileInfo]`, but  then you don't need `"$path\$exe"` just `"$($exe.fullname)"`

Reply:
@TravisEz13 Is `$exe.directoryname` a thing? ... if so, the script could go with ...

```
[Parameter(mandatory=$true)]
[System.IO.FileInfo]
$ExeFileInfo
```

(There would be no separate `$Path` and `$Exe`.)

When I need the path for the ACL, I could use `$($ExeFileInfo.DirectoryName)`.

For the exe, I can use `$($ExeFileInfo.FullName)`.

... and that would still be called with a path, correct? ...

```powershell
.\RegisterService.ps1 
    -Name MyService 
    -DisplayName "My Cool Service" 
    -Description "This is the Sample App service." 
    -ExeFileInfo "c:\svc\SampleApp.exe" 
    -User Desktop-PC\ServiceUser
```

Reply:
basically... I don't think you should call it `ExeFileInfo`.  You shouldn`t include the type in the parameter name as we convert it to the type for you.

Reply:
ok ... I'll stick with `$Exe` then and shown/described with the path.